src/data/store/slices/linkSaturation.slice.ts

```diff
@@ -155,58 +150,112 @@ const linkSaturationSlice = createSlice({
             }
         },
         resetNetworksState: (state) => {
-            state.graphs = {};
+            state.linksPerTemporalEpoch = [];
         },
     },
 });
 
 const updateDRAMLinks = (state: NetworkCongestionState) => {
     const { DRAMBandwidthBytes, CLKHz } = getInitialCLKValues(state);
 
-    Object.values(state.graphs).forEach((graphState) => {
-        Object.values(graphState.links).forEach((linkState: LinkState) => {
-            if (linkState.type === LinkType.DRAM) {
-                linkState.maxBandwidth = DRAMBandwidthBytes / CLKHz;
-                recalculateLinkSaturation(linkState, graphState.totalOps);
-            }
+    Object.values(state.linksPerTemporalEpoch).forEach((epochState) => {
+        Object.entries(epochState.linksPerNodeMap).forEach(([nodeUid, { links, offchipLinkIds }]) => {
+            Object.values(links).forEach((linkState) => {
+                if (linkState.type === LinkType.DRAM) {
+                    linkState.maxBandwidth = DRAMBandwidthBytes / CLKHz;
+                    const { bpc, linkSaturation, nodeSaturation, offChipSaturation } = recalculateLinkSaturation({
+                        link: linkState,
+                        totalOpCycles: epochState.totalOps,
+                        links,
+                        offchipLinkIds,
+                    });
+
+                    linkState.bpc = bpc;
+                    linkState.saturation = linkSaturation;
+                    epochState.linksPerNodeMap[nodeUid].saturation = nodeSaturation;
+                    epochState.linksPerNodeMap[nodeUid].offchipSaturation = offChipSaturation;
+                }
+            });
         });
     });
 };
 
 const updatePCILinks = (state: NetworkCongestionState) => {
     const { PCIBandwidthGBs, CLKHz } = getInitialCLKValues(state);
 
-    Object.values(state.graphs).forEach((graphState) => {
-        Object.values(graphState.links).forEach((linkState: LinkState) => {
-            if (linkState.type === LinkType.PCIE) {
-                linkState.maxBandwidth = PCIBandwidthGBs / CLKHz;
-                recalculateLinkSaturation(linkState, graphState.totalOps);
-            }
+    Object.values(state.linksPerTemporalEpoch).forEach((epochState) => {
+        Object.entries(epochState.linksPerNodeMap).forEach(([nodeUid, { links, offchipLinkIds }]) => {
```

Comment:
`linksPerNodeMap` -> `linksPerNode`? i  actually dislike linksPerNode - there is room for improvent here